### PR TITLE
Allow options to be supplied to IocConstruct

### PR DESCRIPTION
**Pull request by martijn00**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)

Feature

### :arrow_heading_down: What is the current behavior?

Not able to add arguments to IoCConstruct

### :new: What is the new behavior (if this is a feature change)?

You can pass them in now.

### :boom: Does this PR introduce a breaking change?

It breaks the interface, but should not affect any user code.

### :memo: Links to relevant issues/docs

Fixes https://github.com/MvvmCross/MvvmCross/issues/534
Part of: https://github.com/MvvmCross/MvvmCross/issues/2069

### :thinking: Checklist before submitting

- [x] All projects build
- [x] Follows style guide lines ([code style guide](https://github.com/MvvmCross/MvvmCross#code-style-guidelines))
- [ ] Relevant documentation was updated ([docs style guide](https://www.mvvmcross.com/documentation/contribute/mvvmcross-docs-style-guide))
- [x] Rebased onto current develop

### CODE REVIEW

File: MvvmCross/IoC/MvxIoCContainer.cs [hunk 1]
```diff
@@ -584,13 +606,17 @@ protected virtual void InjectProperties(object toReturn)
             _propertyInjector?.Inject(toReturn, _options.PropertyInjectorOptions);
         }
 
-        protected virtual List<object> GetIoCParameterValues(Type type, ConstructorInfo firstConstructor)
+        protected virtual List<object> GetIoCParameterValues(Type type, ConstructorInfo firstConstructor, IDictionary<string, object> arguments)
```
Comment (martijn00) [inline MvvmCross/IoC/MvxIoCContainer.cs]: Should `arguments`be `arguments  = null`?

Comment (Cheesebaron) [reply MvvmCross/IoC/MvxIoCContainer.cs]: I don't think it matters at all.

Comment (MKuckert) [reply MvvmCross/IoC/MvxIoCContainer.cs]: Internal method. `arguments` is always passed.

File: Projects/Playground/Playground.Wpf/Playground.Wpf.csproj [hunk 2]
```diff
@@ -160,6 +160,10 @@
       <Project>{49f965f5-5824-4bbe-94ea-0178f848abdb}</Project>
       <Name>MvvmCross</Name>
     </ProjectReference>
+    <ProjectReference Include="..\..\..\MvvmCross\MvvmCross.Platforms.Wpf.csproj">
```
Comment (nickrandolph) [inline Projects/Playground/Playground.Wpf/Playground.Wpf.csproj]: Should these fixes for WPF sample be pulled out to a separate PR so they can be applied to 6.0.1?

Comment (Cheesebaron) [reply Projects/Playground/Playground.Wpf/Playground.Wpf.csproj]: It is already there: https://github.com/MvvmCross/MvvmCross/blob/develop/Projects/Playground/Playground.Wpf/Playground.Wpf.csproj#L163

Comment (nickrandolph) [reply Projects/Playground/Playground.Wpf/Playground.Wpf.csproj]: yep, so I guess this comes down to how we need to manage PRs/Commits across multiple banches. Do we need to rebase the 6.1 branch to pick up the commit to develop, so that the changes don't appear in this PR?

Comment (Cheesebaron) [reply Projects/Playground/Playground.Wpf/Playground.Wpf.csproj]: Either that or just get it in through this PR.

Comment (nickrandolph) [reply Projects/Playground/Playground.Wpf/Playground.Wpf.csproj]: True, although I guess my point is really to try to define what the process is, so that it is clear to everyone.

Comment (martijn00) [reply Projects/Playground/Playground.Wpf/Playground.Wpf.csproj]: I did the rebase on 6.1 and this branch, so i'm not sure how this ended up here.

Comment (nickrandolph) [reply Projects/Playground/Playground.Wpf/Playground.Wpf.csproj]: Do we need to rebase 6.1 onto develop again?

Comment (martijn00) [reply Projects/Playground/Playground.Wpf/Playground.Wpf.csproj]: Done
